.github/workflows/test-and-lint.yml

```diff
@@ -0,0 +1,115 @@
+        with:
+          components: rustfmt, clippy
+
+      # --- Minimum supported rust version --- 
+      - name: Set up MSRV checker
+        uses: taiki-e/install-action@cargo-hack
+
+      # --- Python toolchain setup for mypy ---
+      - name: Set up Python for linting
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11' # Use a specific version for linting
+
+      # --- Rust checks ---
+      - name: Check formatting
+        run: cargo fmt --all -- --check
+
+      - name: Run Clippy
+        run: cargo clippy --all --all-targets --all-features -- -D warnings
+
+      - name: Check MSRV (Minimum Supported Rust Version)
+        run: cargo hack check --rust-version --workspace --all-targets --ignore-private
+
+      # --- Python mypy checks ---
+      - name: Install and run mypy
+        run: |
+          python -m pip install --upgrade pip mypy
+          mypy --show-column-numbers --hide-error-context .
+
+  test:
+    name: Test on ${{ matrix.os }} with Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    needs: lint # start after the 'lint' job has succeeded.
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Cache so it builds faster.
+      - name: Cache Cargo dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Cache pip dependencies
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/requirements-dev.txt') }}
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install maturin pytest
+          # If you have a requirements-dev.txt, you can install it here.
+          # if [ -f requirements-dev.txt ]; then pip install -r requirements-dev.txt; fi
+
+      - name: Run Rust tests
+        run: cargo test --verbose
+
+      - name: Build wheel and run Python tests
+        # 'maturin develop' builds the Rust extension and installs it for the current Python env.
+        # 'pytest' then runs the tests located in your 'python_bindings' directory or similar.
+        run: |
+          maturin develop
+          pytest python_bindings/tests.py
```

Comment:
You need to enable a virtual environment, otherwise maturin gets pretty angry at you.